pkg/resource/aws/aws_route_table_association_ext.go

```diff
@@ -1,14 +1,42 @@
 package aws
 
-import "fmt"
+import (
+	"bytes"
+	"fmt"
 
-func (r *AwsRouteTableAssociation) String() string {
-	assoc := fmt.Sprintf("Table: %s", *r.RouteTableId)
+	"github.com/hashicorp/terraform/helper/hashcode"
+)
+
+func (r *AwsRouteTableAssociation) CreateIdHash() string {
+	var buf bytes.Buffer
+	if r.RouteTableId != nil && *r.RouteTableId != "" {
+		buf.WriteString(fmt.Sprintf("%s-", *r.RouteTableId))
+	}
+	if r.GatewayId != nil && *r.GatewayId != "" {
+		buf.WriteString(fmt.Sprintf("%s-", *r.GatewayId))
+	}
+	if r.SubnetId != nil && *r.SubnetId != "" {
+		buf.WriteString(fmt.Sprintf("%s-", *r.SubnetId))
+	}
+	if buf.Len() > 0 {
+		return fmt.Sprintf("rtassociation-%d", hashcode.String(buf.String()))
+	}
+	return ""
+}
+
+func (r *AwsRouteTableAssociation) Attrs() map[string]string {
+	attrs := make(map[string]string)
+	if idHash := r.CreateIdHash(); idHash != "" {
```

Comment:
same here. Printing or getting attribute should not create new attributes

Reply:
CreateIdHash() deleted.